hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/management/JpaTargetManagement.java

```diff
@@ -566,7 +569,7 @@ private List<JpaTarget> findTargetsByInSpecification(final Collection<String> co
             ConcurrencyFailureException.class }, maxAttempts = Constants.TX_RT_MAX, backoff = @Backoff(delay = Constants.TX_RT_DELAY))
     public List<Target> assignTag(final Collection<String> controllerIds, final long tagId) {
         final List<JpaTarget> allTargets = targetRepository
-                .findAll(AccessController.Operation.UPDATE, TargetSpecifications.byControllerIdWithTagsInJoin(controllerIds));
+                .findAll(TargetSpecifications.byControllerIdWithTagsInJoin(controllerIds));
```

Comment:
I removed the update check to first verify if all requested targets are accessible at all. In case one of the controllerIds is not readable/visible, we will answer the request with an EntityNotFoundException.class

What is indeed missing is the check for update access in a second step. I will introduce it after the read check:

    if (allTargets.size() < controllerIds.size()) {
        throw new EntityNotFoundException(Target.class, controllerIds,
                allTargets.stream().map(Target::getControllerId).toList());
    }
    // new here
    targetRepository.getAccessController().ifPresent(
            acm -> acm.assertOperationAllowed(AccessController.Operation.UPDATE, allTargets));